src/api/API.js

```diff
@@ -0,0 +1,114 @@
+import EventEmitter from 'events';
+import axios from 'axios';
+
+import Annotation from '../Annotation';
+import { getHeaders } from '../util';
+
+class API extends EventEmitter {
+    /**
+     * Generates a rfc4122v4-compliant GUID, from
+     * http://stackoverflow.com/questions/105034/create-guid-uuid-in-javascript.
+     *
+     * @return {string} UUID for annotation
+     */
+    static generateID() {
+        /* eslint-disable */
+        return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, (c) => {
+            var r = (Math.random() * 16) | 0,
+                v = c == 'x' ? r : (r & 0x3) | 0x8;
+            return v.toString(16);
+        });
+        /* eslint-enable */
+    }
+
+    /**
+     * [constructor]
+     *
+     * @param {Object} data - Annotation Service data
+     * @return {API} API instance
+     */
+    constructor(data) {
+        super();
+        this.apiHost = data.apiHost;
+        this.fileId = data.fileId;
+        this.headers = getHeaders({}, data.token);
+        this.canAnnotate = data.canAnnotate;
```

Comment:
what does `canAnnotate` mean for an API class? Does it get used in API requests?

Reply:
What was the resolution on this comment? Having these annotation-specific properties in a base API object seems odd.

Reply:
canAnnotate is not needed here. Removed it